k8sclient/k8sclient.go

```diff
@@ -686,6 +692,16 @@ func tryLoadK8sPodDefaultNetwork(k8sArgs *types.K8sArgs, conf *types.NetConf, ku
 		return nil, logging.Errorf("tryLoadK8sPodDefaultNetwork: more than one default network is specified: %s", netAnnot)
 	}
 
+	if networks[0].Name == "" {
```

Comment:
I am not sure I follow. Why do we require an empty name?

Reply:
Maybe I miss something. I am not sure I understand what you say.

Let's get to basics.

What is the goal of the `defaultNetwork` annotation? My understanding is that it tells Multus which network should be used instead of default one configured as the primary delegate in its configuration file. This is achieved by listing the corresponding network name in `defaultNetwork` annotation. This annotation was added in: https://github.com/intel/multus-cni/pull/174/

The current limitation of this approach that you try to address is that right now `defaultNetwork` annotation value format is a single network name which user may be interested in passing additional arguments / attributes to the CNI plugin.

The goal of your PR is to make `defaultNetwork` annotation support JSON format that would include the name of the default network but also additional attributes. (This behavior reflects handling of `networks` attribute which may be of two formats: simple comma separated list as well as a JSON list.)

In the example I see added by this PR in `doc/configuration.md`, you only list additional attributes (`mac`) without core attributes (`name`, `namespace`). In this case, it's not possible for Multus to determine which network should be used as default for the pod.

Perhaps your PR adds ability to pass additional attributes to the default cluster network. But by doing so, it also replaces ability to choose a different network to use as default for a pod. Which is a problem.

Is there something I miss in my explanation?

Reply:
@booxter thanks for the comment I will try to answer

> Perhaps your PR adds ability to pass additional attributes to the default cluster network. But by doing so, it also replaces ability to choose a different network to use as default for a pod. Which is a problem.

this PR adds ability to pass additional attributes in 3 cases.
1. configure only extra parameter (use the multus.conf delegate)
https://github.com/intel/multus-cni/pull/268/files#diff-bdea1124c0b7de33d6bff38302fad2aaR570
2. configure only extra parameter (use the multus.conf clusterNetwork)
https://github.com/intel/multus-cni/pull/268/files#diff-bdea1124c0b7de33d6bff38302fad2aaR605
3. configure default network and extra parameter
https://github.com/intel/multus-cni/pull/268/files#diff-bdea1124c0b7de33d6bff38302fad2aaR641

continue to support the network name in default network annotation
https://github.com/intel/multus-cni/blob/e64db39f44939774d8d082edc4fd5782ba4c187e/k8sclient/k8sclient_test.go#L495

please tell me if this comment answer you question